tests/unit/util/test_file_converter.py

```diff
@@ -0,0 +1,165 @@
+"""Unit tests for functions in file_converter.py"""
+
+import os
+
+import pytest
+import pandas as pd
+import pathlib
+import json
+import yaml
+import datetime as dt
+
+from clean_air.util import file_converter as fc
+
+
+@pytest.fixture()
+def excel_filepath(sampledir):
+    filepath = os.path.join(sampledir, "test_data",
+                            "metadata_form_responses.xlsx")
+    return filepath
+
+
+@pytest.fixture()
+def tmp_output_path(tmp_path):
+    tmp_output_path = tmp_path / "tmp_output_path"
+    tmp_output_path.mkdir()
+    return tmp_output_path
+
+
+@pytest.fixture()
+def saved_json(excel_filepath, tmp_output_path):
+    json_fname = tmp_output_path / "form_response0.json"
+    fc.convert_excel_to_json(excel_filepath, tmp_output_path)
+    saved_json = json.load(json_fname.open())
+    return saved_json
+
+
+@pytest.fixture()
+def saved_yaml(excel_filepath, tmp_output_path):
+    yaml_fname = tmp_output_path / "form_response0.yaml"
+    fc.convert_excel_to_yaml(excel_filepath, tmp_output_path)
+    saved_yaml = yaml.safe_load(yaml_fname.open())
+    return saved_yaml
+
+
+def test_read_excel_data(excel_filepath):
+    """Test that excel files are read and converted successfully to
+    temporary dataframe objects"""
+    temp_df = fc.read_excel_data(filepath=excel_filepath)
+    assert isinstance(temp_df, pd.DataFrame)
+
+
+def test_slice_data(excel_filepath):
+    """Test that data from temporary dataframes (read from excel files) are
+    split into single dataframes for each row of data.  Test excel file has
+    three rows, so should be split into three separate files here."""
+    # First, read and slice the test file:
+    temp_df = fc.read_excel_data(filepath=excel_filepath)
+    sliced_data = fc.slice_data(temp_df)
+    # Now check that three separate files have been generated:
+    assert len(sliced_data) == 3
+
+
+def test_json_reformat_chemicals(saved_json):
+    """Test that the single excel entry for each chemical is reformatted
+    into three seperate lines representing 'name', 'shortname' and
+    'chart' information."""
+    len_chems = len(saved_json["pollutants"])
+    for i in range(len_chems):
```

Comment:
`for chem in saved_json["pollutants"]:` is the more pythonic way of looping over lists. Or `for i, chem in enumerate(saved_json["pollutants"]):` if an index is needed.

Reply:
I don't see what's wrong with this as it is.  I am happy to change "i" to be "chem" instead, but other than that I think you're just nitpicking now.

Nevertheless, I value your opinion and advice, so I will amend this to your tastes.